main/src/cgeo/geocaching/log/LogCacheActivity.java

```diff
@@ -215,9 +217,20 @@ private void showErrorLoadingAdditionalData() {
     }
 
     private void initializeTrackablesAction() {
-        if (Settings.isTrackableAutoVisit()) {
-            for (final TrackableLog trackable : trackables) {
-                trackable.action = LogTypeTrackable.VISITED;
+        for (final TrackableLog trackable : trackables) {
+            if (trackableState == null) { // initial call
+                if (Settings.isTrackableAutoVisit()) {
+                    trackable.action = LogTypeTrackable.VISITED;
+                }
+            } else { // refresh view
+                final Integer tbStateId = trackableState.getInt(trackable.trackCode);
+                if (tbStateId > 0) { // found in saved list
+                    trackable.action = LogTypeTrackable.getById(tbStateId);
+                } else { // not found
+                    if (Settings.isTrackableAutoVisit()) {
```

Comment:
I don't like the duplicated condition. Alternative loop:
```
        for (final TrackableLog trackable : trackables) {
            if (trackableState != null) { // refresh view
                final int tbStateId = trackableState.getInt(trackable.trackCode);
                if (tbStateId > 0) { // found in saved list
                    trackable.action = LogTypeTrackable.getById(tbStateId);
                    continue;
                }
            }
            if (Settings.isTrackableAutoVisit()) {
                trackable.action = LogTypeTrackable.VISITED;
            }
        }
```

Reply:
I don't like the duplicated condition too. 
Changed.